### PR TITLE
feat: restrict Causes users from seeing MG and AD signups

### PR DESCRIPTION
**Pull request by eddieferrer**
ACK-219
ACK-220

I tested all instances of this. 

We're now doing this in quite a few places. A future improvement would be better to centralize all this business logic somewhere. Maybe straight from the API? my.userAccount.isEligibleForMonthlyGood or maybe a local resolver of some sort?

### CODE REVIEW

File: src/components/MonthlyGood/AlreadySubscribedNotice.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template functional>
 	<div v-if="!props.onetime">
 		<h1 class="tw-text-center tw-text-h2 tw-mb-2">
-			You're already signed up for Monthly Good
+			You're already signed up for a subscription
```
Comment (eddieferrer) [inline src/components/MonthlyGood/AlreadySubscribedNotice.vue:4]: I changed the wording here to be able to reuse this messaging for users who have MG or Causes

File: src/components/Settings/SubscriptionsSettingsCards.vue [hunk 2]
```diff
@@ -154,13 +155,13 @@ export default {
...
 			this.isLegacySubscriber = legacySubs.length > 0;
-			this.mySubscriptions = data?.mySubscriptions?.values ?? [];
-			const causesSubscriptions = this.mySubscriptions.filter(
+
+			const modernSubscriptions = data?.mySubscriptions?.values ?? [];
+			this.hasModernSub = modernSubscriptions.length !== 0;
+			const causesSubscriptions = modernSubscriptions.filter(
 				subscription => subscription.category.subscriptionType === 'CAUSES'
 			);
-			this.hasActiveCauseSubscription = !!causesSubscriptions.find(
```
Comment (eddieferrer) [inline src/components/Settings/SubscriptionsSettingsCards.vue:161]: cleaned up this logic a little bit.

File: src/pages/LandingPages/MGCovid19/MGCovid19.vue [hunk 3]
```diff
@@ -50,8 +50,6 @@ export default {
 	},
 	data() {
 		return {
-			isExperimentActive: true,
-			isMonthlyGoodSubscriber: false,
```
Comment (eddieferrer) [inline src/pages/LandingPages/MGCovid19/MGCovid19.vue:54]: This data is not used in this template. I also did not modify `MGCovidHero` which does have a MG signup. But I don't think we are going to be using that landing page again.

Comment (mcstover) [reply src/pages/LandingPages/MGCovid19/MGCovid19.vue:54]: Oh man I sure hope we don't!

**Review by mcstover — APPROVED** [review]
Thank so you much!

**Pull request merged**